pkg/v3/resource/namespace/delete.go

```diff
@@ -0,0 +1,15 @@
+package namespace
+
+import (
+	"context"
+
+	"github.com/giantswarm/operatorkit/controller"
+)
+
+func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
+	return nil
+}
```

Comment:
I understand `giantswarm` namespace is never deleted? I think it should be canceled in `GetCurrentState` and commented why is that.

Check https://github.com/giantswarm/azure-operator/blob/b819000b00ba0c82a405b5bdf19eae59f996b7f5/service/controller/v1/resource/service/current.go#L22-L27 as an example. This saves some K8s calls and logs.

Reply:
Yes doesn't make sense to delete the namespace since the cluster itself is being deleted. Thanks this is nice and will save some cycles.

I had to add a new accessor function for getting the object metadata. Since this is a generic resource shared by all 3 providers.

@kopiczko @fgimenez PTAL